unit/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -96,6 +96,20 @@ expr_to_smt_conversion_test_environmentt::convert(const exprt &expression) const
     object_size_function.make_application);
 }
 
+TEST_CASE("\"array_typet\" to smt sort conversion", "[core][smt2_incremental]")
+{
+  auto test =
+    expr_to_smt_conversion_test_environmentt::make(test_archt::x86_64);
+
+  const auto array_type =
+    array_typet{signedbv_typet{8}, from_integer(8, c_index_type())};
+  INFO("Type being converted: " + array_type.pretty(2, 0));
+  const auto conversion_result = convert_type_to_smt_sort(array_type);
+  CHECK(
+    conversion_result ==
+    smt_array_sortt{smt_bit_vector_sortt{64}, smt_bit_vector_sortt{8}});
+}
+
```

Comment:
The question is which spec? The specification for ArraysEx allows sorts of Arrays of any sort of index and value. However we are targeting the logic `QF_AUFBV`. This logic only supports arrays with bit vector index and element sorts. Therefore this PR is currently missing the piece of work for converting booleans to bit vectors in the case where we have an cbmc array of booleans.

TLDR We should be putting arrays of nonbitvector type into the conversions and checking that we always get arrays of bit vectors out.

Reply:
With the current changes CBMC will allow multi-dimensional array and boolean array support.
The SMT backend (z3 and cvc5) support that, except z3 on ubuntu 18.04 (z3 version 4.4.0) that rejects it as it is not SMTlib 2.6.

My suggestion is to keep this as is and without additional tests (that would fail on ubuntu 18.04).